Automattic/jetpack
# Search: Add essential scaffolding for package

#### Changes proposed in this Pull Request:
Adds essential scaffolding to the Search package.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
Ensure that the changes look reasonable and that the new `pnpm` script commands all work as expected in `projects/packages/search`:
- `pnpm clean-all` - clean up builds and dependencies
- `pnpm build` - would only echo 'Not yet implemented.'
- `pnpm build-development` - would only echo a couple of 'Not yet implemented.'
- `pnpm build-production` - would only echo a couple of 'Not yet implemented.'
- `pnpm test` - runs `jest`, currently no tests
- `pnpm test-size` - would only echo 'Not yet implemented.'
